### PR TITLE
Docs: Marker support to Avi objects

### PR DESCRIPTION
**Pull request by akshayhavile**

### CODE REVIEW

**Review by monotosh-avi — APPROVED** [review]

**Comment by akshayhavile** [comment]
build aki

**Comment by akshayhavile** [comment]
build ako

File: docs/objects.md [hunk 1]
```diff
@@ -328,6 +328,195 @@ The formula to derive the pool's PKIprofile is as follows:
...
+
+Passthrough pool will be marked as follows:
+
+```
+clustername : <clustername>
+Host        : <fqdn>
+ServiceName : <servicename>
+```
+
+#### Advance L4 VS
```
Comment (chauhanshubham) [inline docs/objects.md:409]: The gateway VSes created as part of regular `servicesAPI: true` workflow is not covered here. We should probably replace those with `Advance L4` objects, probably to `L4 Gateway VS, L4 Gateway Pool..`

File: docs/objects.md [hunk 2]
```diff
@@ -328,6 +328,195 @@ The formula to derive the pool's PKIprofile is as follows:
...
+
+Advance L4 Vs will be marked as follows:
+
+```
+clustername      : <clustername>
+Namespace        : <namespace>
+GatewayName      : <gatewayname>
+```
+
+#### Advance L4 Pool Name
```
Comment (chauhanshubham) [inline docs/objects.md:419]: I think "Name" is not required.

File: docs/objects.md [hunk 3]
```diff
@@ -328,6 +328,195 @@ The formula to derive the pool's PKIprofile is as follows:
...
+
+Advance L4 pool name will be marked as follows:
```
Comment (chauhanshubham) [inline docs/objects.md:421]: same here, is `name` required here?

File: docs/objects.md (hunk 2 above)
Comment (akshayhavile) [reply docs/objects.md:419]: While deriving Adv L4 VS and Adv pool name, these fields are used. So we are marking these objects with these labels.

File: docs/objects.md (hunk 3 above)
Comment (akshayhavile) [reply docs/objects.md:421]: Same as above

File: docs/objects.md (hunk 1 above)
Comment (akshayhavile) [reply docs/objects.md:409]: Did not understand comment. Can you please elaborate more?

File: docs/objects.md [hunk 4]
```diff
@@ -328,6 +328,195 @@ The formula to derive the pool's PKIprofile is as follows:
 ```
 pkiprofilenam = poolName + "-pkiprofile"
 ```
+### Markers for Avi Objects
+
+With Avi controller version 20.1.5, Avi objects can be labelled with markers. Each marker is pair of `key` : `<value1>`, `<value2>`. Each object can have multiple markers. Markers can be used to group objects at Avi controller side and provide permission to users to access these objects. Avi controller support markers to all license types from 20.1.6.
```
Comment (sudswasavi) [inline docs/objects.md:333]: Each marker is a

Comment (sudswasavi) [reply docs/objects.md:333]: Markers are used to group/filter objects on the Avi Controller and can be utilised to provide user access permissions using the GRBAC feature on the Avi Controller.

Avi controller supports markers in all controller license types from 20.1.6.

File: docs/objects.md [hunk 5]
```diff
@@ -328,6 +328,195 @@ The formula to derive the pool's PKIprofile is as follows:
...
+
+With AKO version 1.5.1, AKO add markers to Avi objects. Each objects will be marked with fields of ingress/routes such as namespace, fqdn of ingress/route, service name of ingress/route etc. This will help in identifying object associate with specific ingress/route at controller side. Markers will be added to objects in SNI and EVH deployment modes.
```
Comment (sudswasavi) [inline docs/objects.md:335]: From AKO 1.5.1, markers would be added to all the AKO created Avi Objects.

Each object will be marked with fields such as OpenShift/Kubernetes namespaces to which the object belongs, fqdn present in the ingress/route objects, service names present in the ingress/route objects.

**Review by sudswasavi — COMMENTED** [review]
Few comments.

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Review by sudswasavi — APPROVED** [review]
@chauhanshubham shall we merge this?

**Review by chauhanshubham — APPROVED** [review]

**Pull request merged**